Upgrade python versions to allow 3.11 installation and fix current pip installation failure

jmfrank63 opened the pull request:
Currently tested on a Mac M1, this alongside the piper PR should fix
`pip install piper-tts`, which already created three issues.

Upgraded several python versions.
Windows and Linux, not tested yet.

jmfrank63 committed:
upgrade versions

jmfrank63 committed:
Fix lints

jmfrank63 committed:
Fix codepoints

jmfrank63 committed:
Set CMP0135 policy new

jmfrank63 committed:
Fix makefile

jmfrank63 committed:
fixing wip

jmfrank63 committed:
Windows compiles and pip installs

jmfrank63 committed:
Add files

jmfrank63 committed:
Use dir copy

jmfrank63 committed:
Fix copying, now works

jmfrank63 commented:
This is now at a point, where some conversation should take place. Currently, MacOS and Windows are working and roughly tested. Linux will follow, but I do not expect the same chore as with Windows.
Both this and the `piper` are modified. If you want to run this, set an environment variable `PIPER_PHONEMIZE_REPO` and point it to your fork, so `piper` will load it.

rmcpantoja commented:
> This is now at a point, where some conversation should take place. Currently, MacOS and Windows are working and roughly tested. Linux will follow, but I do not expect the same chore as with Windows. Both this and the `piper` are modified. If you want to run this, set an environment variable `PIPER_PHONEMIZE_REPO` and point it to your fork, so `piper` will load it.

Hi @jmfrank63, 
How did you manage to do a wheel on Windows?

I tried through GitHub actions but got errors:
https://github.com/rmcpantoja/piper-phonemize/blob/master/.github/workflows/main.yml

jmfrank63 commented:
I don't know exactly what kind of errors you are experiencing, but it might be the Unicode that is not understood by the Microsoft C-compiler. I changed the utf-8 characters to the hex representation and added conditional compilation for windows. But I must have made a mistake because it broke other languages except English, which now produce pure nonsense. Haven't had the time to look into this yet.

rmcpantoja commented:
> I don't know exactly what kind of errors you are experiencing, but it might be the Unicode that is not understood by the Microsoft C-compiler. I changed the utf-8 characters to the hex representation and added conditional compilation for windows. But I must have made a mistake because it broke other languages except English, which now produce pure nonsense. Haven't had the time to look into this yet.

You can see the full log [here](https://actions-results-receiver-production.githubapp.com/rest/runs/fef7c2d5-1be5-4f7c-96e8-beac473a618e/logs?filename=logs_21095067562.zip). It says it can't find espeak-ng/include, maibe I'm doing pip wheel in an wrong path...
